### ISSUE TITLE
implement GO-CAM Alliance Web widget

### ISSUE THREAD

**Issue by ValWood** [pombase/website#2169]
So far, we have 1 production model for testing.
Should be more soon.
http://noctua.geneontology.org/workbench/noctua-visual-pathway-editor/?model_id=gomodel%3A66187e4700001573

code
https://github.com/geneontology/wc-gocam-viz

**Comment by kimrutherford** [comment]
Is this instead of esyN?

**Comment by ValWood** [comment]
Kind of. esyN  was more of a 'precursor' but the functionality would still be useful (the uses are slightly overlapping)
can chat about it...

**Comment by kimrutherford** [comment]
Prototype!

https://desktop.kmr.nz/gene/SPBC2F12.08c

![image](https://github.com/pombase/website/assets/90474/4bc16098-fe07-4674-b81f-671688819e4c)

**Comment by ValWood** [comment]
Amazing!!!!!!!!!!

**Comment by pgaudet** [comment]
That was incredibly fast!!!

**Comment by ValWood** [comment]
CC @PCarme

**Comment by ValWood** [comment]
That might be because of the Noctua outage for updating. That doesn't seem to be a good way to do things if all the GO-CAM views totally disappear. Most resources would update and switch the data rather than update the live version.
@pgaudet, do you know if there is a long-term plan to reduce Notua release downtime? This could be very frustrating for users.

**Comment by pgaudet** [comment]
Yes, Seth would like to implement the Noctua back-end in such a way that we can do 'hot' updates. This will take a little while though, many parts need changing.

**Comment by ValWood** [comment]
Don't all the APIs plug into the Noctua database though?

**Comment by ValWood** [comment]
nice!

**Comment by kimrutherford** [comment]
The two other pathways are on the dev site now.

http://dev.pombase.kmr.nz/gene/SPAC24B11.11c
http://dev.pombase.kmr.nz/gene/SPAC21E11.03c

![image](https://github.com/pombase/website/assets/90474/f78476a7-3569-4c04-b0e1-2da181f8f55a)

**Comment by ValWood** [comment]
Fantastic!

**Comment by kimrutherford** [comment]
Currently we're storing the gene ID to GO-CAM ID mapping is a manually updated file:
```
pombe-embl/supporting_files/production_gocam_id_mapping.tsv
```

We should switch to using the API to find pombe GO-CAMs:
https://app.swaggerhub.com/apis-docs/geneontology/gosparql/2.2#/Users/get_users__orcid__models

We'll need to wait for the pombe models to be available via the API.

If there's always going to be a lag we might need to keep using the SVN file as a supplement.  There's no problem using both.

**Comment by ValWood** [comment]
Is it out of action? The GOCAMs don't load right now but other widgets do.

BTW I made another pathway production at the weekend
http://noctua.geneontology.org/workbench/noctua-visual-pathway-editor/?model_id=gomodel%3A662af8fa00000408
it needs extending, but the part that is there is correct (I was checking to see if would appear automatically)

**Comment by ValWood** [comment]
ignore, it was slow

**Comment by kimrutherford** [comment]
> I made another pathway production at the weekend

OK, I've added that to the mapping file.  It should be on the gene pages in the morning.

**Comment by ValWood** [comment]
OK, sorry I should have twigged that. I will add to the mapping file once in production.

**Comment by kimrutherford** [comment]
I just checked and we already have two genes with more than one GO-CAM

 - pcr1 / SPAC21E11.03c 
 - atf1 / SPBC29B5.01 

are both in both these models:
 - 66187e4700001781
 - 66187e4700003150

so we need a way to select which model to view on the gene page.

Also I think it makes sense to have a "Show larger view" link to show a dedicated page for the widget, like we do for protein features.

**Comment by kimrutherford** [comment]
> Also I think it makes sense to have a "Show larger view" link to show a dedicated page for the widget, like we do for protein features.

Now I've though about it, there's a bit of a difference.  For the protein features we need a larger view page for each gene but we'll just need a single page for each GO-CAM.  I'm working on that today.

**Comment by ValWood** [comment]
Good idea!

**Comment by kimrutherford** [comment]
> Also I think it makes sense to have a "Show larger view" link to show a dedicated page for the widget, like we do for protein features.

Prototype here: https://desktop.kmr.nz/gene/SPBC2F12.08c

![image](https://github.com/pombase/website/assets/90474/e4c28163-94c2-492d-b797-15ae8c2cbf9b)

![image](https://github.com/pombase/website/assets/90474/77e67f20-d1de-4eb2-bd42-bc885e64102f)

**Comment by ValWood** [comment]
Nice!

**Comment by kimrutherford** [comment]
The current prototype is on the dev site: http://dev.pombase.kmr.nz/gene/SPBC29B5.01

![image](https://github.com/pombase/website/assets/90474/4b563dde-6055-4ccc-9b87-f72c6740c056)

I'm working on replacing the GO-CAM IDs in the selector with model titles so that it's more friendly.  Hopefully we can use the GO API to retrieve the titles and other details.

We should also add an info icon and maybe a link to the GO-CAM docs: https://geneontology.org/docs/gocam-overview/

**Comment by PCarme** [comment]
I just noticed that interactions of type "indirectly positively/negatively regulates" are not rendered in the widget, just like "constitutively upstream" links. 

![image](https://github.com/pombase/website/assets/163283897/64881a8a-95dd-4d34-bf3d-c10d186018f4)

**Comment by ValWood** [comment]
Its the same here too
http://noctua.geneontology.org/workbench/noctua-alliance-pathway-preview/?model_id=gomodel%3A66187e4700001781

so its an upstream problem. Can you report it here:
https://github.com/geneontology/noctua/issues
thanks,

**Comment by kimrutherford** [comment]
We should add a link to a help/documentation page and maybe a one-line description above the diagram, for clarity.

**Comment by kimrutherford** [comment]
I've changed the icon.  I didn't like the previous version:

![image](https://github.com/pombase/website/assets/90474/ace24a16-fc4c-44d6-937f-c5886b2d1800)

**Comment by ValWood** [comment]
the diagrams are currently not visible on dev?

**Comment by ValWood** [comment]
ignore I thought coq* model would be visible, must be a time lag

**Comment by ValWood** [comment]
I don't think it needs much of a description. Maybe

change
GO-CAM model 66187e4700003150
to
GO-CAM pathway model for:

and the process will be in the deader underneath.
We can link "GO-CAM" to our internal documentation which will link out to the broader GO documentation.

I will write a draft for that.

**Comment by ValWood** [comment]
Draft documentation


GO-CAMs, or Gene Ontology Causal Activity Models, are a framework within the Gene Ontology (GO) project that aims to represent complex biological processes in a structured, formal, and computable manner. GO-CAMs extend the traditional GO annotations by incorporating causal relationships between activities (e.g., molecular functions) and linking them to biological processes and cellular components. This allows for the representation of how gene products interact and function in a biological context. This involves specifying the entities (genes, proteins, etc.), their activities, and the causal relationships between them. GO-CAMs are represented as graphs where nodes represent molecular activities and edges represent causal relationships. This graphical representation enables the capture of complex interactions and pathways.

At PomBAse, we have begun representing biological pathways as GO-CM Models and displaying them on the relevant gene and process pages.

Pathways with production GO-CAM models in GO-CAM <LINK to a list of production models , this could be added later>

For further information see the Gene Ontology GO-CAM overview https://geneontology.org/docs/gocam-overview/ documentation.

**Comment by kimrutherford** [comment]
> change
> GO-CAM model 66187e4700003150
> to
> GO-CAM pathway model for:

We'll need to allow for the ID (66187e4700003150) for now because some genes are in multiple pathways so the ID is a drop-down.

**Comment by kimrutherford** [comment]
> <LINK to a list of production models , this could be added later>

Would that be a link to Noctua?  It's not very user friendly?
I've reported a bug in linking to a list of production models:
  - https://github.com/geneontology/noctua/issues/895

Maybe we could have a table of models on the documentation page?  It could be automatically generated.

**Comment by ValWood** [comment]
I agree, I was thinking we could maintain a local list

**Comment by kimrutherford** [comment]
I've added the draft documentation and a link to it from the top of the widget:
  - http://dev.pombase.kmr.nz/gene/SPBC29B5.01
  - http://dev.pombase.kmr.nz/documentation/go-cam-pathway-models

**Comment by ValWood** [comment]
Looks good.

slightly rev ised text:


GO-CAMs, or Gene Ontology Causal Activity Models, are a framework within the Gene Ontology (GO) project that aims to represent complex biological processes in a structured, formal, and computable manner. GO-CAMs extend the traditional GO annotations by incorporating causal relationships between activities (e.g., molecular functions) and linking them to biological processes and cellular components. This approach allows for a detailed representation of how gene products interact and function within a biological context.
GO-CAMs involve specifying entities such as genes and proteins, their activities, and the causal relationships between them. These models are represented as graphs, where nodes depict molecular activities and edges denote causal relationships. This graphical representation captures complex interactions and pathways, providing a comprehensive view of biological processes.

**Comment by kimrutherford** [comment]
Thanks, I've put that text in.

**Comment by kimrutherford** [comment]
We could consider linking to the model at Amigo:
https://amigo.geneontology.org/amigo/model/66187e4700001781
from the larger view page.

**Comment by kimrutherford** [comment]
> Maybe we could have a table of models on the documentation page? It could be automatically generated.

I've had a first attempt at this.  The table is generated from the data in Chado.  As you can see, there is some missing information that needs to be added.

http://dev.pombase.kmr.nz/documentation/go-cam-pathway-models

![image](https://github.com/pombase/website/assets/90474/8e356905-2493-4566-86ce-ade4d279e1f9)

**Comment by ValWood** [comment]
Let's discuss on Tuesday.
I  think we need to include the title of the model.

For example 
662af8fa00000499 
|soprenoid biosynthetic process
ergosterol biosynthetic process

for this the model  title is "ergosterol biosynthetic process"  (soprenoid biosynthetic process is included because isoprenoids are a precursor to ergosterol)

and  for 
silent heterochromatin formation  model some of the other processes like "siRNA-independent heterochromatin formation" are sub-processes which are part of "heterochromatin formation, but cannot have this parentage in the ontology because it isn't always true  are 'sub processes'

...although looking at the current model these have changed, because we are refining the existing annotations as we go.
The differences are due to a synchronization issue.

I guess we need to wait until the  Noctua update to see these revisions...

but it needs to be clear  what the overall model is.

**Comment by kimrutherford** [comment]
> I think we need to include the title of the model.

I'll add the titles once they're available from GO.  We're waiting on API bug fixes.

**Comment by ValWood** [comment]
> We could consider linking to the model at Amigo:

yep

**Comment by kimrutherford** [comment]
I've moved the comments about the GO-CAM table in the docs to:
  - pombase/website#2183

I forgot we had a separate issue for it.

**Comment by kimrutherford** [comment]
From tomorrow, the selector will show the model titles.  Something like this:

--------------------------

![image](https://github.com/pombase/website/assets/90474/0259c692-37d9-44cf-a3cd-52a31296f079)

-----------------------------------

![image](https://github.com/pombase/website/assets/90474/a37d7b52-5f64-4815-bfc1-7c77980bc389)

**Comment by kimrutherford** [comment]
> From tomorrow, the selector will show the model titles. Something like this:

Here's some examples:

http://dev.pombase.kmr.nz/gene/SPAC664.01c
http://dev.pombase.kmr.nz/term/GO:0140719

**Comment by ValWood** [comment]
I think we can close this ticket, and open new tickets for any changes?

**Comment by kimrutherford** [comment]
> I think we can close this ticket, and open new tickets for any changes?

I looked back through the comments and noticed that I haven't done this:

> We could consider linking to the model at Amigo:

I'll add that then close the issue.

**Comment by kimrutherford** [comment]
>> We could consider linking to the model at Amigo:
>
> I'll add that then close the issue.

That's done now and will be on the dev site in 20 minutes or so.
